client/state/plugins/premium/actions.js

```diff
@@ -0,0 +1,251 @@
+			name: plugin.name,
+			key: plugin.key,
+			status: {
+				start: false,
+				install: null,
+				activate: null,
+				config: null,
+				done: false,
+			},
+			error: null,
+		}
+	} );
+}
+
+function install( site, plugin, dispatch ) {
+	Dispatcher.handleViewAction( {
+		type: 'INSTALL_PLUGIN',
+		action: 'INSTALL_PLUGIN',
+		site: site,
+		plugin: plugin,
+		data: null,
+		error: null
+	} );
+	wpcom.pluginsInstall( site.ID, plugin.slug, ( error, data ) => {
+		if ( error ) {
+			if ( error.name === 'PluginAlreadyInstalledError' ) {
+				update( site, plugin, dispatch );
+			} else {
+				dispatch( {
+					type: PLUGIN_SETUP_ERROR,
+					siteId: site.ID,
+					slug: plugin.slug,
+					error,
+				} );
+				Dispatcher.handleServerAction( {
+					type: 'RECEIVE_INSTALLED_PLUGIN',
+					action: 'INSTALL_PLUGIN',
+					site: site,
+					plugin: plugin,
+					data: null,
+					error: error
+				} );
+			}
+			return;
+		}
+
+		dispatch( {
+			type: PLUGIN_SETUP_ACTIVATE,
+			siteId: site.ID,
+			slug: plugin.slug,
+		} );
+
+		activate( site, data, dispatch );
+	} );
+};
+
+function update( site, plugin, dispatch ) {
+	wpcom.pluginsUpdate( site.ID, plugin.id, ( error ) => {
+		if ( error ) {
+			dispatch( {
+				type: PLUGIN_SETUP_ERROR,
+				siteId: site.ID,
+				slug: plugin.slug,
+				error,
+			} );
+			Dispatcher.handleServerAction( {
+				type: 'RECEIVE_INSTALLED_PLUGIN',
+				action: 'INSTALL_PLUGIN',
+				site: site,
+				plugin: plugin,
+				data: null,
+				error: error
+			} );
+			return;
+		}
+
+		dispatch( {
+			type: PLUGIN_SETUP_ACTIVATE,
+			siteId: site.ID,
+			slug: plugin.slug,
+		} );
+
+		activate( site, plugin, dispatch );
+	} );
+};
+
+function activate( site, plugin, dispatch ) {
+	wpcom.pluginsActivate( site.ID, plugin.id, ( error, data ) => {
+		if ( error && error.name !== 'ActivationErrorError' ) {
+			dispatch( {
+				type: PLUGIN_SETUP_ERROR,
+				siteId: site.ID,
+				slug: plugin.slug,
+				error,
+			} );
+			Dispatcher.handleServerAction( {
+				type: 'RECEIVE_INSTALLED_PLUGIN',
+				action: 'INSTALL_PLUGIN',
+				site: site,
+				plugin: plugin,
+				data: null,
+				error: error
+			} );
+
+			return;
+		}
+
+		if ( error && error.name === 'ActivationErrorError' ) {
+			// Technically it failed, but we need the plugin info in `data`.
+			data = plugin;
+		}
+
+		dispatch( {
+			type: PLUGIN_SETUP_CONFIGURE,
+			siteId: site.ID,
+			slug: plugin.slug,
+		} );
+		Dispatcher.handleServerAction( {
+			type: 'RECEIVE_INSTALLED_PLUGIN',
+			action: 'INSTALL_PLUGIN',
+			site: site,
+			plugin: plugin,
+			data: data,
+			error: null
+		} );
+
+		autoupdate( site.ID, data );
+		configure( site, data, dispatch );
+	} );
+};
+
+function autoupdate( siteId, plugin ) {
+	wpcom.enableAutoupdates( siteId, plugin.id );
+};
+
+function configure( site, plugin, dispatch ) {
+	setTimeout( () => {
+		dispatch( {
+			type: PLUGIN_SETUP_FINISH,
+			siteId: site.ID,
+			slug: plugin.slug,
+		} );
+	}, 1500 );
```

Comment:
Since it's an action on the user's site it could take some time. That said, this will be going away in another PR shortly after this gets merged (when the real configuration step is implemented).